### PR TITLE
Remove non async eviction

### PR DESCRIPTION
**Pull request by calvinjia**
Removes the deprecated synchronous eviction code path.
Previously, Alluxio supported both synchronous and asynchronous eviction. Synchronous eviction was more straightforward but ultimately less efficient because each client request would evict a small amount of data (enough to satisfy the current request space call). Asynchronous eviction allows the evictor to use more information and do a more efficient batch eviction.

### CODE REVIEW

File: core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java [hunk 1]
```diff
@@ -357,47 +311,23 @@ public void moveBlock(long sessionId, long blockId, BlockStoreLocation oldLocati
...
           }
-          return;
         }
-        // Failed to create a temp block, so trigger Evictor to make some space.
-        // NOTE: a successful {@link freeSpaceInternal} here does not ensure the subsequent
-        // allocation also successful, because these two operations are not atomic.
-        freeSpaceInternal(sessionId, result.getBlockSize(), newLocation, Mode.GUARANTEED);
+        return;
       }
-      // TODO(bin): We are probably seeing a rare transient failure, maybe define and throw some
```
Comment (ZacBlanco) [inline core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java]: keep TODO?

File: tests/src/test/java/alluxio/server/tieredstore/SpecificTierWriteIntegrationTest.java [hunk 2]
```diff
@@ -45,8 +45,8 @@
  * Integration tests for writing to various storage tiers.
  */
 public class SpecificTierWriteIntegrationTest extends BaseIntegrationTest {
-  private static final int CAPACITY_BYTES = Constants.KB;
-  private static final int FILE_SIZE = CAPACITY_BYTES;
+  private static final int CAPACITY_BYTES = Constants.KB * 10;
```
Comment (ZacBlanco) [inline tests/src/test/java/alluxio/server/tieredstore/SpecificTierWriteIntegrationTest.java]: Why are we bumping these?

File: tests/src/test/java/alluxio/server/tieredstore/SpecificTierWriteIntegrationTest.java [hunk 3]
```diff
@@ -45,8 +45,8 @@
...
+  private static final int FILE_SIZE = CAPACITY_BYTES / 10;
```
Comment (ZacBlanco) [inline tests/src/test/java/alluxio/server/tieredstore/SpecificTierWriteIntegrationTest.java]: similarly, the next person who comes along might not realize that the `10` on capacity and file size are related. Maybe make it a single constant variable and mult/divide by that?

File: tests/src/test/java/alluxio/server/tieredstore/TierPromoteIntegrationTest.java [hunk 4]
```diff
@@ -43,7 +43,7 @@
 
 @RunWith(Parameterized.class)
 public class TierPromoteIntegrationTest extends BaseIntegrationTest {
-  private static final int CAPACITY_BYTES = Constants.KB;
+  private static final int CAPACITY_BYTES = 10 * Constants.KB;
```
Comment (ZacBlanco) [inline tests/src/test/java/alluxio/server/tieredstore/TierPromoteIntegrationTest.java]: Why bump?

File: tests/src/test/java/alluxio/server/tieredstore/SpecificTierWriteIntegrationTest.java [hunk 5]
```diff
@@ -178,32 +181,45 @@ public void allTierWrite() throws Exception {
...
 
   @Test
   public void allTierWriteWithEviction() throws Exception {
-    writeFileAndCheckUsage(0, FILE_SIZE, 0, 0);
-    writeFileAndCheckUsage(1, FILE_SIZE, FILE_SIZE, 0);
-    writeFileAndCheckUsage(2, FILE_SIZE, FILE_SIZE, FILE_SIZE);
-    writeFileAndCheckUsage(-1, FILE_SIZE, FILE_SIZE, FILE_SIZE);
-    writeFileAndCheckUsage(-2, FILE_SIZE, FILE_SIZE, FILE_SIZE);
-    writeFileAndCheckUsage(-3, FILE_SIZE, FILE_SIZE, FILE_SIZE);
+    int unreservedTier1 = CAPACITY_BYTES / 10 * 7;
```
Comment (ZacBlanco) [inline tests/src/test/java/alluxio/server/tieredstore/SpecificTierWriteIntegrationTest.java]: Someone might not understand what all of this math is for, could you write a comment explaining what we are doing here, or extract into some aptly-named variables? Same for the rest of the tests in this file

**Comment by ZacBlanco** [comment]
Also, could you provide a short description as to what the benefit of removing synchronous eviction does for Alluxio in terms of performance/simplicity? You could put it in the original PR description so it is included in the commit message

**Comment by calvinjia** [comment]
@ZacBlanco Updated, thanks.

File: tests/src/test/java/alluxio/server/tieredstore/SpecificTierWriteIntegrationTest.java (hunk 2 above)
Comment (calvinjia) [reply tests/src/test/java/alluxio/server/tieredstore/SpecificTierWriteIntegrationTest.java]: When file size = capacity bytes, it is difficult to test eviction (all or nothing). This also isn't the expected case in normal usages of Alluxio, we expect each worker to be able to store more than one file/block.

File: core/server/worker/src/main/java/alluxio/worker/block/SpaceReserver.java [hunk 6]
```diff
@@ -134,14 +125,6 @@ private synchronized void reserveSpace() {
                 reservedSpace, tierAlias, e.getMessage());
           }
         }
-      } else {
```
[the remark this answers is not in the document]
Comment (calvinjia) [reply core/server/worker/src/main/java/alluxio/worker/block/SpaceReserver.java]: `UpdateStorageInfo` will fail if this is not true.

File: core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java [hunk 7]
```diff
@@ -311,38 +285,18 @@ public void abortBlock(long sessionId, long blockId) throws BlockAlreadyExistsEx
...
-        }
-        // Failed to create a temp block, so trigger Evictor to make some space.
-        // NOTE: a successful {@link freeSpaceInternal} here does not ensure the subsequent
-        // allocation also successful, because these two operations are not atomic.
-        freeSpaceInternal(sessionId, additionalBytes, requestResult.getSecond(), Mode.GUARANTEED);
+    LOG.debug("requestSpace: sessionId={}, blockId={}, additionalBytes={}", sessionId, blockId,
+        additionalBytes);
+    RetryPolicy retryPolicy = new TimeoutRetry(FREE_SPACE_TIMEOUT_MS, EVICTION_INTERVAL_MS);
+    while (retryPolicy.attempt()) {
+      Pair<Boolean, BlockStoreLocation> requestResult =
```
[the remark this answers is not in the document]
Comment (calvinjia) [reply core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java]: Unfortunately no

File: core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java [hunk 8]
```diff
@@ -311,38 +285,18 @@ public void abortBlock(long sessionId, long blockId) throws BlockAlreadyExistsEx
...
+          requestSpaceInternal(blockId, additionalBytes);
+      if (requestResult.getFirst()) {
+        return;
       }
-      // TODO(bin): We are probably seeing a rare transient failure, maybe define and throw some
```
[the remark this answers is not in the document]
Comment (calvinjia) [reply core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java]: Updated, thanks.

File: core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java (hunk 1 above)
Comment (calvinjia) [reply core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java]: Updated, thanks.

File: tests/src/test/java/alluxio/server/tieredstore/SpecificTierWriteIntegrationTest.java (hunk 3 above)
Comment (calvinjia) [reply tests/src/test/java/alluxio/server/tieredstore/SpecificTierWriteIntegrationTest.java]: I've changed some of the naming to make the relationships more explicit.

File: tests/src/test/java/alluxio/server/tieredstore/TierPromoteIntegrationTest.java (hunk 4 above)
Comment (calvinjia) [reply tests/src/test/java/alluxio/server/tieredstore/TierPromoteIntegrationTest.java]: See comment for the other one

File: tests/src/test/java/alluxio/server/tieredstore/SpecificTierWriteIntegrationTest.java (hunk 5 above)
Comment (calvinjia) [reply tests/src/test/java/alluxio/server/tieredstore/SpecificTierWriteIntegrationTest.java]: Good point, I've clarified this.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Pull request merged**